File: src/main/java/me/miki/shindo/management/mods/ModCategory.java
package me.miki.shindo.management.mods;

import lombok.Getter;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.utils.animation.ColorAnimation;
import me.miki.shindo.utils.animation.simple.SimpleAnimation;

public enum ModCategory {
    ALL(TranslateText.ALL),
    PLAYER(TranslateText.PLAYER),
    RENDER(TranslateText.RENDER),
    HUD(TranslateText.HUD),
    WORLD(TranslateText.WORLD),
    OTHER(TranslateText.OTHER);

    private final TranslateText nameTranslate;

    @Getter
    private final ColorAnimation textColorAnimation;

    @Getter
    private final SimpleAnimation backgroundAnimation;

    ModCategory(TranslateText nameTranslate) {
        this.nameTranslate = nameTranslate;
        this.backgroundAnimation = new SimpleAnimation();
        this.textColorAnimation = new ColorAnimation();
    }

    public String getName() {
        return nameTranslate.getText();
    }

}

File: src/main/kotlin/me/miki/shindo/management/mods/ModCategory.kt
package me.miki.shindo.management.mods

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.ui.animation.v2.value.ColorAnimation
import me.miki.shindo.ui.animation.v2.value.SimpleAnimation

@Suppress("UNUSED")
enum class ModCategory(
    private val nameTranslate: TranslateText,
) {
    ALL(TranslateText.ALL),
    PLAYER(TranslateText.PLAYER),
    RENDER(TranslateText.RENDER),
    HUD(TranslateText.HUD),
    WORLD(TranslateText.WORLD),
    OTHER(TranslateText.OTHER),
    ;

    val textColorAnimation = ColorAnimation()
    val backgroundAnimation = SimpleAnimation()

    fun getName(): String = nameTranslate.getText()
}
